apps/enphasesolar/enphase_solar.star

```diff
@@ -0,0 +1,174 @@
+    client_secret = config.str("client_secret")
+    api_key = config.str("api_key")
+
+    if not all([access_token, refresh_token, client_id, client_secret, api_key]):
+        return render_msg("Missing credential information.")
+
+    # check if it is initial invocation by check "init" flag in the cache
+    init = cache.get("init")
+    if init == None:
+        cache.set("access_token", access_token, ttl_seconds = TTL_SECONDS)
+        cache.set("refresh_token", refresh_token)
+        cache.set("init", "1")
+
+    # check access token if it needs to be refreshed
+    access_token = cache.get("access_token")
+    if access_token == None:
+        request_refresh_token(cache.get("refresh_token"), client_id, client_secret)
+
+    # Get "energy_today"
+    engery_cached = cache.get("energy_today")
+    if engery_cached == None:
+        status, energy_today = get_system_stats(api_key)
+        if status == 200:
+            cache.set("energy_today", energy_today, ttl_seconds = 60)
+        elif status == 401:
+            request_refresh_token(cache.get("refresh_token"), client_id, client_secret)
+            return render_msg("Token just refreshed wait for next call.")
+        else:
+            return render_msg("Unable to get system stats, status code: {}".format(status))
+    else:
+        print("Hit! Displaying cached data.")
+        energy_today = engery_cached
+
+    msg = "{} kWh energy generated today".format(humanize.float("#.##", float(energy_today) / 1000))
+    return render_msg(msg)
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Toggle(
+                id = "switch",
+                name = "Switch On/Off",
+                desc = "Switch the app on or off",
+                icon = "gear",
+                default = False,
+            ),
+            schema.Text(
+                id = "api_key",
+                name = "API Key",
+                desc = "API key for the Enphase API. Follow https://developer-v4.enphase.com/docs/quickstart.html#create_account.",
+                icon = "key",
+            ),
+            schema.Text(
+                id = "client_id",
+                name = "Client ID",
+                desc = "The client id of Enphase application",
+                icon = "key",
+            ),
+            schema.Text(
+                id = "client_secret",
+                name = "Client Secret",
+                desc = "The client secret of Enphase application",
+                icon = "key",
+            ),
+            schema.Text(
+                id = "access_token",
+                name = "Access Token",
+                desc = "Follow the App README to generate the access_token",
+                icon = "key",
+            ),
+            schema.Text(
+                id = "refresh_token",
+                name = "Refresh Token",
+                desc = "Follow the App README to generate the refresh_token",
```

Comment:
These descriptions are unfortunately not visible to end users in the iOS/Android app. Could you update the the app description in the manifest to hold the relevant information for users to get going with the app?